var/spack/repos/builtin/packages/mvapich/package.py

```diff
@@ -287,6 +288,11 @@ def configure_args(self):
         else:
             args.append("--disable-registration-cache")
 
+        ld = ""
+        for path in itertools.chain(self.compiler.extra_rpaths, self.compiler.implicit_rpaths()):
```

Comment:
We received a bug report from @nhanford that the rpath's were not correctly being added to MVAPICH's compiler wrappers. This was an attempt to resolve that. Perhaps Nate can weigh in if we went about this in a method different from what he was thinking?

Reply:
@haampie We're seeing similar issues to #8670 and the proposed solution happens to resemble #8687. Perhaps there is a better way to go about this?

Reply:
We run `filter_compiler_wrappers` on the installed `mpicc`, etc., because the Spack wrappers cannot be executed outside of Spack's build environment, so you want your `mpicc` to execute with the *real* compiler.  If the real compiler needed extra RPATHs in Spack, it likely needs them outside Spack, so it makes a lot of sense (to me) that we'd make the wrappers add these paths.

Spack-installed MPIs are *not* expected to be run just by Spack -- you want them to work with `module load` too.